marshmallow_recipe/generics.py

```diff
@@ -0,0 +1,119 @@
+import dataclasses
+import types
+import typing
+from typing import Annotated, Any, Generic, TypeAlias, TypeVar, Union, get_args, get_origin
+
+_GenericAlias: TypeAlias = typing._GenericAlias  # type: ignore
+
+TypeLike: TypeAlias = type | TypeVar | types.UnionType | types.GenericAlias | _GenericAlias
+TypeVarMap: TypeAlias = dict[TypeVar, TypeLike]
+ClassTypeVarMap: TypeAlias = dict[TypeLike, TypeVarMap]
+FieldsTypeVarMap: TypeAlias = dict[str, TypeVarMap]
+FieldsClassMap: TypeAlias = dict[str, TypeLike]
+FieldsTypeMap: TypeAlias = dict[str, TypeLike]
+
+
+def get_fields_type_map(t: TypeLike) -> FieldsTypeMap:
+    origin = get_origin(t) or t
+    if not dataclasses.is_dataclass(origin):
+        return {}
+
+    class_type_var_map = get_class_type_var_map(t)
+    fields_class_map = get_fields_class_map(t)
+    return {
+        f.name: build_subscripted_type(f.type, class_type_var_map.get(fields_class_map[f.name], {}))
+        for f in dataclasses.fields(origin)
+    }
+
+
+def get_fields_class_map(t: TypeLike) -> FieldsClassMap:
+    origin = get_origin(t) or t
+    if not dataclasses.is_dataclass(origin):
+        return {}
+
+    names: dict[str, dataclasses.Field] = {}
+    result: FieldsClassMap = {}
+
+    mro = origin.__mro__  # type: ignore
+    for cls in (*mro[-1:0:-1], origin):
+        if not dataclasses.is_dataclass(cls):
+            continue
+        for field in dataclasses.fields(cls):
+            if names.get(field.name) != field:
+                names[field.name] = field
+                result[field.name] = cls
+
+    return result
+
+
+def build_subscripted_type(t: TypeLike, type_var_map: TypeVarMap) -> TypeLike:
+    if isinstance(t, TypeVar):
+        return build_subscripted_type(type_var_map[t], type_var_map)
+
+    origin = get_origin(t)
+    if origin is Union or origin is types.UnionType:
+        return Union[*(build_subscripted_type(x, type_var_map) for x in get_args(t))]  # type: ignore
```

Comment:
Pyright highlights `*` as unsupported for Python 3.10 in this case of usage but as I see there are no Python 3.10 in tests matrix. 

Should I support this behaviour for Python 3.10 or can I increase Python version for Pyright up to 3.11?

The same thing for `Annotated` type on the next line

Reply:
I have updated this branch and removed `# type: ignore` for this case

also I noticed that Pyright does not notify about redundant `# type: ignore` as mypy do